src/gui/symbols/icon_properties_widget.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ *    Copyright 2018 Kai Pastor
+ *
+ *    This file is part of OpenOrienteering.
+ *
+ *    OpenOrienteering is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    OpenOrienteering is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with OpenOrienteering.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "icon_properties_widget.h"
+
+#include <QSpinBox>
+#include <QFormLayout>
+#include <QHBoxLayout>
+#include <QImageReader>
+#include <QImageWriter>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QPushButton>
+#include <QLineEdit>
+
+#include "core/symbols/symbol.h"
+#include "gui/file_dialog.h"
+#include "gui/util_gui.h"
+#include "gui/symbols/symbol_setting_dialog.h"
+#include "util/backports.h"
+
+namespace OpenOrienteering {
+
+namespace {
+
+QString pngFileFilter()
+{
+	static const QString filter_template(QLatin1String("%1 (%2)"));
+	QStringList filters = {
+	    filter_template.arg(::OpenOrienteering::IconPropertiesWidget::tr("PNG"), QLatin1String("*.png")),
+	    ::OpenOrienteering::IconPropertiesWidget::tr("All files (*.*)")
+	};
+	return filters.join(QLatin1String(";;"));
+}
+
+
+}  // namespace
+
+
+
+IconPropertiesWidget::IconPropertiesWidget(Symbol* symbol, SymbolSettingDialog* dialog)
+: QWidget(dialog)
+, dialog(dialog)
+, symbol(symbol)
+{
+	auto layout = new QFormLayout();
+	setLayout(layout);
+	
+	layout->addRow(Util::Headline::create(tr("Default icon")));
+	
+	default_icon_display = new QLabel();
+	layout->addRow(default_icon_display);
+	
+	default_icon_size_edit = Util::SpinBox::create(22, 256, tr("px"));
+	layout->addRow(tr("Width:"), default_icon_size_edit);
```

Comment:
What is the purpose of the icon size change spin box?

Reply:
For now, that's the only way to change the size of the default icon when you want to save it for customization. We can use a [QInputDialog](http://doc.qt.io/qt-5/qinputdialog.html#details) instead. This change will create sequence of two dialogs  for saving the default icon (1. file dialog, 2. size dialog).

PS: Should there be an extra Save button for the default icon?

Reply:
The spinbox is fine from my POV. It is the intended workflow which is unclear to me at the moment.

Regarding buttons, my first thought was having "Populate custom icon from generated one". That way all operations can be restricted to the custom icon. In the end I also think that size of both icons can be locked and controlled by only one spinbox.

Summary of my proposal:
1) display both generated and custom icon in the dialog
2) provide possibility to load and save custom icon for external modifications
3) provide possibility to copy generated icon into custom one (for later editing)
4) control icon size by only one spinbox, rescale custom icon upon size changes
5) future - provide integrated icon editor

Remark 1 - Custom icons get (down)scaled and color-reduced when exported into OCD.
Remark 2 - Vector based custom icons might be worth a shot. Graphic primitives can be limited to line and Bézier curve + line width and color settings.

Reply:
ad 1) already implemented  
ad 2) already implemented  
ad 3) now an explicit button  
ad 4) not sure about this one. We rescale on display, everything else is better handled in graphics software. However, I relabeled the spinbox to "Preview width", and moved the "width" lines to before the icons.  
ad 5) future anyway.